lib/fiddle.rb

```diff
@@ -10,98 +10,72 @@
 require 'fiddle/version'
 
 module Fiddle
-  if WINDOWS
-    # Returns the last win32 +Error+ of the current executing +Thread+ or nil
-    # if none
-    def self.win32_last_error
-      if RUBY_ENGINE == 'jruby'
-        errno = FFI.errno
-        errno == 0 ? nil : errno
-      else
-        Thread.current[:__FIDDLE_WIN32_LAST_ERROR__]
-      end
+  case RUBY_ENGINE
+  when 'jruby'
+    def self.last_error
+      FFI.errno.nonzero?
     end
 
-    # Sets the last win32 +Error+ of the current executing +Thread+ to +error+
-    def self.win32_last_error= error
-      if RUBY_ENGINE == 'jruby'
-        FFI.errno = error || 0
-      else
-        Thread.current[:__FIDDLE_WIN32_LAST_ERROR__] = error
-      end
-    end
-
-    # Returns the last win32 socket +Error+ of the current executing
-    # +Thread+ or nil if none
-    def self.win32_last_socket_error
-      if RUBY_ENGINE == 'jruby'
-        errno = FFI.errno
-        errno == 0 ? nil : errno
-      else
-        Thread.current[:__FIDDLE_WIN32_LAST_SOCKET_ERROR__]
-      end
+    def self.last_error= error
+      FFI.errno = error || 0
     end
 
-    # Sets the last win32 socket +Error+ of the current executing
-    # +Thread+ to +error+
-    def self.win32_last_socket_error= error
-      if RUBY_ENGINE == 'jruby'
-        FFI.errno = error || 0
-      else
-        Thread.current[:__FIDDLE_WIN32_LAST_SOCKET_ERROR__] = error
+    if WINDOWS
+      class << self
+        alias win32_last_error last_error
+        alias win32_last_error= last_error=
+        alias win32_last_socket_error last_error
+        alias win32_last_socket_error= last_error=
       end
     end
-  end
-
-  # Returns the last +Error+ of the current executing +Thread+ or nil if none
-  def self.last_error
-    if RUBY_ENGINE == 'jruby'
```

Comment:
Mmh, actually `FFI.last_error{,=}` should also use FFI.errno on TruffleRuby, it seems somehow I missed this, maybe because there are no tests for this?

Orthogonal to this PR, so feel free to merge this first.

Reply:
Ah it wasn't caught because of https://github.com/ruby/fiddle/blob/d5fd917123ca3cff9a976a4505848589cf822e49/test/fiddle/test_function.rb#L128-L131

Reply:
=> https://github.com/ruby/fiddle/pull/173